src/utils/HTTPTransport.ts

```diff
@@ -0,0 +1,88 @@
+enum METHODS {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  PATCH = 'PATCH',
+  DELETE = 'DELETE',
+}
+
+type TRequestData = Record<string, string | number>;
+
+type TRequestOptions = {
+  method?: METHODS
+  headers?: Record<string, string>
+  timeout?: number
+  data?: unknown
+};
+
+function queryStringify(data: TRequestData) {
+  if (!data) return '';
+  return Object.entries(data).reduce((acc, [key, value], index, arr) => {
+    return `${acc}${key}=${value}${index < arr.length - 1 ? '&' : ''}`;
+  }, '?');
```

Comment:
Это решение выглядит довольно сложным, я предложу более простое решение:
```suggestion
  return "?"  + Object
   .entries(data)
   .map(([key, value]) => `${key}=${value}`)
   .join("&")
```

Тут осуществляется 2 прохода по массиву, вместо одного, зато код в разы понятнее и его будет легче поддерживать